cosima_cookbook/querying.py

```diff
@@ -7,28 +7,51 @@
 import logging
 import os.path
 import pandas as pd
-from sqlalchemy import func, distinct
+from sqlalchemy import func, distinct, select, or_
 import warnings
 import xarray as xr
 
 from . import database
-from .database import NCExperiment, NCFile, CFVariable, NCVar
+from .database import NCExperiment, NCFile, CFVariable, NCVar, Keyword
 
 
 class VariableNotFoundError(Exception):
     pass
 
-def get_experiments(session):
+def get_experiments(session, experiment=True, keywords=None, all=False, **kwargs):
```

Comment:
Argument docs? These are selecting the columns?

Reply:
Yeah good point. In general I would probably prefer to return the keywords as a tuple field in the  `pandas.DataFrame` and deal with the filtering in panda-land. Unfortunately it isn't obvious to me that multi-way comparison between a list of desired keywords and a tuple of keywords in a cell in a `pandas.DataFrame` is straightforward.

Reply:
I've also realised this is an `OR` operation on the keywords, but I think I want `AND`: the more keywords specified the narrower the list of returned experiments.

Not sure the best way to achieve that. Multiple `.in_` filters for each keyword?

Reply:
In some way, I could imagine it'd be cool to compose the filter: all experiment with `'cosima' and ('iaf-something' or 'ryf9091')`, for example. But, maybe that'd be easier to do at a different level. In the meantime, I'll see what's on offer for doing the set intersection test.

Reply:
Yeah that would be useful, and I did think there would be legit use cases for the current behaviour, but couldn't see a simple way to achieve it. If `pandas` had some nice machinery for this it would make sense to return a keywords tuple in a single column, but it doesn't seem a great fit.

I've folded the `keyword` filtering into the `get_experiments` function for convenience. The cleanest way for it to work would be return an object to which commands could be chained, as `xarray` does. That would require some considerable retooling.